net/oic/include/oic/port/mynewt/stream.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * stream.h: Utility code for adaptors that implement the CoAP-over-TCP
+ * protocol (Bluetooth, TCP/IP, etc.).
+ */
+
+#ifndef H_MYNEWT_TCP_
+#define H_MYNEWT_TCP_
+
+#include "os/mynewt.h"
+#include "oic/oc_ri.h"
+#include "oic/oc_ri_const.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Indicates whether a transport-specific endpoint matches the provided
+ * description.
+ */
+typedef bool oc_stream_ep_match(const void *ep, void *ep_desc);
```

Comment:
Any reason why only ep is const?

Reply:
Thanks.  I agree `ep_desc` should also be const here.  I also think it should be const in the "fill" callback.  I have fixed both.